core/src/main/java/org/infinispan/notifications/cachelistener/cluster/ClusterListenerReplicateCallable.java

```diff
@@ -101,26 +101,27 @@ public Void call() throws Exception {
                   // We only need to use the post cache event filter when a filter is provided, since the remote cluster listener
                   // will ignore pre events anyways.  This is mostly so a user filter is not notified of the pre event.
                   CacheEventFilter<K, V> actualFilter = null;
-                  if (filter instanceof CacheEventFilterConverter) {
-                     final CacheEventFilterConverter<K, V, ?> filterConverter = (CacheEventFilterConverter<K, V, ?>) filter;
-                     actualFilter = new AbstractCacheEventFilterConverter<K,V, Object>() {
-                        @Override
-                        public Object filterAndConvert(K key, V oldValue, Metadata oldMetadata, V newValue, Metadata newMetadata, EventType eventType) {
-                           if (eventType.isPreEvent()) {
-                              return null;
-                           }
-                           return filterConverter.filterAndConvert(key, oldValue, oldMetadata, newValue, newMetadata, eventType);
-                        }
-                     };
-                  } else if (filter != null) {
-                     actualFilter = new CacheEventFilter<K, V>() {
-                        @Override
-                        public boolean accept(K key, V oldValue, Metadata oldMetadata, V newValue, Metadata newMetadata, EventType eventType) {
-                           return !eventType.isPreEvent() && filter.accept(key, oldValue, oldMetadata, newValue, newMetadata, eventType);
-                        }
-                     };
-                  }
-                  cacheNotifier.addListener(listener, actualFilter, converter);
+// TODO the wrapping of the filter prevents it to be detected as an IndexedFilter
```

Comment:
Like I said it should work without it, but that means filter and converter will be fired for both pre and post events.  The listener will in turn throw away the pre event so it should work fine, but requires an additional invocation.  We could do something like what JDBC does with unwrap maybe?